src/System.Management.Automation/engine/Modules/AnalysisCache.cs

```diff
@@ -665,7 +667,7 @@ private void Cleanup()
                 }
             }
 
-            if (removedSomething)
+            if (_saveCacheToDisk && removedSomething)
```

Comment:
I think this check can be removed - it would be redundantly checked in `QueueSerialization` and it will rarely be true, so the extra code isn't saving anything in the normal case.